### ISSUE TITLE
Allow calling methods by a literal string variable

### ISSUE THREAD

**Issue by danog** [vimeo/psalm#8538]
<?php
class a {
    public static function a(): int { return 0; }
	public static function b(): string { return "test"; }
}

$c = random_int(0, 1) ? 'a' : 'b';
/** @psalm-trace $c */;
$c = a::$c();
/** @psalm-trace $c */;

**Comment by orklah** [comment]
I'm quite surprised it doesn't work. It's not even an issue with the `'a'|'b'` thing. Even with `'a'` it fails. It shouldn't be difficult as I think something close to that must be already working for callable-arrays resolution

**Comment by AndrolGenhald** [comment]
Related: https://github.com/vimeo/psalm/issues/5743#issuecomment-1026250945

> We don't currently resolve literal strings to callables: https://psalm.dev/r/13a7a514f3. Not sure if that's an intentional choice or not.

**Comment by orklah** [comment]
I don't think this was intentional, more like a nice to have feature that never made it.

Now that we had an RFC to declare what we can and can't do with callable-arrays, it makes even more sense to support all that is still relevant
